stempy/image.cpp

```diff
@@ -36,6 +42,55 @@ STEMValues calculateSTEMValues(uint16_t data[], int offset,
   return stemValues;
 }
 
+#ifdef VTKm
+STEMValues calculateSTEMValuesParallel(uint16_t data[], int offset,
+                                       int numberOfPixels,
+                                       uint16_t brightFieldMask[],
+                                       uint16_t darkFieldMask[],
+                                       uint32_t imageNumber)
+{
+  STEMValues stemValues;
+  stemValues.imageNumber = imageNumber;
+
+  auto keysBright = vtkm::cont::make_ArrayHandle(brightFieldMask,
+                                                 numberOfPixels);
+  auto keysDark = vtkm::cont::make_ArrayHandle(darkFieldMask, numberOfPixels);
+  auto input = vtkm::cont::make_ArrayHandle(&data[offset], numberOfPixels);
+
+  vtkm::cont::ArrayHandle<uint16_t> uniqueKeys;
+  vtkm::cont::ArrayHandle<uint16_t> sums;
+
+  // Portals so we may view the values
+  auto uniqueKeysPortal = uniqueKeys.GetPortalConstControl();
```

Comment:
You should get the portals after the algorithm has executed, not before.
You should presume that all data parallel algorithm are executing in an async manner and that the algorithm will execute intil another algorithm execution, or `ArrayHandle::GetPortalConstControl` is called.

Reply:
Oh okay. Will do.